### ISSUE TITLE
What's the difference between **reload** and **gracefulReload**?

### ISSUE THREAD

**Issue by andrei-cocorean** [Unitech/pm2#2025]
Hi, I was wondering why there are 2 commands for reloading. For my server I have a shutdown function to wait for all connections to close:

``` javascript
const server = http.createServer(reqListener)
server.listen()

function shutdown () {
  server.close(() => {
    console.log('server closed')
    process.exit()
  })
}
```

The [cluster mode docs](http://pm2.keymetrics.io/docs/usage/cluster-mode/) say that I should listen for a shutdown message:

``` javascript
process.on('message', msg => {
  if (msg === 'shutdown') {
    shutdown()
  }
})
```

and then exec `$ pm2 gracefulReload app` in the console.

But I can also listen for SIGINT `process.on('SIGINT', shutdown)` and then do `$ pm2 reload app` in the console.

As far as I can tell both approaches do the same thing. Am I missing some case covered by gracefulReload?
## 

pm2 v1.0.2
node v4.4.0

**Comment by soyuka** [comment]
Reload will fallback to restart after a specified delay (`GRACEFUL_LISTEN_TIMEOUT` see [here](https://github.com/Unitech/pm2/blob/dc995569097c2c5b118e22b43a1299990563e47d/lib/samples/sample-conf.js#L29)). After this delay, it'll force the script to restart even if it hasn't finished to process leftovers. 

If you want to know more, please take a look at the [`Reload` code source](https://github.com/Unitech/pm2/blob/98c4e6e012e013862d5161446b6495fab8816bfc/lib/God/Reload.js).

**Comment by dytsai** [comment]
I believe the problem here is the documentation. It suggests listening to SIGINT at the linke http://pm2.keymetrics.io/docs/usage/signals-clean-restart for graceful reload but mentions process "shutdown" message here at http://pm2.keymetrics.io/docs/usage/pm2-doc-single-page. SIGINT is sent whenever app is stopped/restarted/deleted/reloaded/gracefulReloaded but shutdown message is only sent when doing graceful reload, which by default will wait for 8000ms before sending SIGINT to the process.

**Comment by andrei-cocorean** [comment]
Thanks for the clarification. It seems like the shutdown message covers just a single use case, while catching SIGINT covers all. I expect any cleanup code to be executed whenever the server shuts down, regardless if it's part of a reload or a stop. Is there any reason to use gracefulReload at all?

**Comment by Unitech** [comment]
Patch available on pm2@2.1.4 (main):

```bash
$ npm install pm2 -g
$ pm2 update
```
